lib/dangermattic/plugins/labels_checker.rb

```diff
@@ -0,0 +1,41 @@
+# frozen_string_literal: true
+
+module Danger
+  # Plugin to check for labels in a PR.
+  class LabelsChecker < Plugin
+    DEFAULT_DO_NOT_MERGE_LABEL = 'Do Not Merge'
+    DEFAULT_LABELS_WARNING = 'Please add the required labels for this project.'
+    DEFAULT_REQUIRED_LABELS = [/.*\S.*/].freeze
+
+    # Checks if a PR is missing labels or is marked as 'do not merge'.
+    #
+    # @param do_not_merge_label [String] The label indicating that a merge should not be performed.
+    #   Defaults to DEFAULT_DO_NOT_MERGE_LABEL if not provided.
+    # @param required_labels [Array<String>] The list of labels that must be present for a merge operation to proceed.
+    #   Defaults to an empty array if not provided.
+    # @param required_labels_warning [String] The warning message displayed if the required labels are not present.
+    #   Defaults to DEFAULT_LABELS_WARNING if not provided.
+    #
+    # @return [void]
+    def check(do_not_merge_label: DEFAULT_DO_NOT_MERGE_LABEL, required_labels: DEFAULT_REQUIRED_LABELS,
+              required_labels_warning: DEFAULT_LABELS_WARNING)
+      github_labels = danger.github.pr_labels
+
+      pr_has_all_required_labels = github_labels.all? do |pr_label|
+        required_labels.any? do |required_label|
+          pr_label =~ required_label
+        end
+      end
+
+      if github_labels.empty?
+        warn('PR is missing at least one label.')
+      elsif !pr_has_all_required_labels
+        warn(required_labels_warning)
```

Comment:
Would be nice to print the list of missing required labels, so that the PR author knows what's missing and so that the warning is more actionable, otherwise I think it would feel pretty obscure?
```suggestion
        warn("#{required_labels_warning}\nPR is missing label(s) matching: #{missing_required_labels})
```

Reply:
I thought about that, but showing regexes could be a bit cryptic as well 🤔 (well, it is still better than nothing). But that's the main reason I introduced the `required_labels_warning` parameter as well, so that the expected labels could be set inside a free text message from outside of the plugin, in addition to just providing regexes for validation.

Reply:
Yeah, [hence my other suggestion above](https://github.com/Automattic/dangermattic/pull/7#discussion_r1262747487) which would allow the default message to print the expected regexes if no explicit message was provided, while still letting the called provide an explicit warning message to replace it if they prefer.

Then this part of the code would look like:
```ruby
unless missing_required_labels.empty?
  warn(required_labels_warning || "PR is missing label(s) matching: #{missing_required_labels})
end
```

Reply:
Yes, I [just seen it now](https://github.com/Automattic/dangermattic/pull/7#discussion_r1262792975)! 🙂